### PR TITLE
chore: migrate instructions to learn

### LINKED ISSUE #43836: i18n: Python projects' instructions are not translatable
For these projects the instructions are in the README file inside the boilerplate, as such they are not translatable.

I propose to add a copy of the projects readme inside the project file in the main repo.

List of projects that need it:
### Scientific Computing with Python
- [x] Arithmetic formatter #43854 
- [x] Time Calculator #44152
- [x] Budget App #44125
- [x] Polygon Area Calculator #44152
- [x] Probability Calculator #44152

### Data Analysis with Python
- [X] Mean-Variance-Standard Deviation Calculator #44160
- [X] Demographic Data Analyzer #44160
- [X] Medical Data Visualizer #44160
- [X] Page View Time Series Visualizer #44160
- [X] Sea Level Predictor #44160

### Information Security  
- [x] Port Scanner #44241
- [x] SHA-1 Password Cracker #44241

### Machine Learning with Python 
- [ ] Rock Paper Scissors
- [ ] Cat and Dog Image Classifier
- [ ] Book Recommendation Engine using KNN
- [ ] Linear Regression Health Costs Calculator
- [ ] Neural Network SMS Text Classifier 

---

(Mods feel free to edit as needed)

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] I have tested these changes either locally on my machine, or GitPod.

Closes #43836 

This PR migrates the last of the python project instructions to `/learn`.

### CODE REVIEW

File: curriculum/challenges/english/11-machine-learning-with-python/machine-learning-with-python-projects/book-recommendation-engine-using-knn.md [hunk 1]
```diff
@@ -8,16 +8,49 @@ dashedName: book-recommendation-engine-using-knn
...
 
 We are still developing the interactive instructional content for the machine learning curriculum. For now, you can go through the video challenges in this certification. You may also have to seek out additional learning resources, similar to what you would do when working on a real-world project.
 
+# --instructions--
+
+In this challenge, you will create a book recommendation algorithm using **K-Nearest Neighbors**.
+
+You will use the [Book-Crossings dataset](http://www2.informatik.uni-freiburg.de/~cziegler/BX/). This dataset contains 1.1 million ratings (scale of 1-10) of 270,000 books by 90,000 users.
+
+After importing and cleaning the data, use `NearestNeighbors` from `sklearn.neighbors` to develop a model that shows books that are similar to a given book. The Nearest Neighbors algorithm measures distance to determine the “closeness” of instances.
```
Comment (Sembauke) [inline curriculum/challenges/english/11-machine-learning-with-python/machine-learning-with-python-projects/book-recommendation-engine-using-knn.md:23]:
```suggestion
After importing and cleaning the data, use `NearestNeighbors` from `sklearn.neighbors` to develop a model that shows books that are similar to a given book. The Nearest Neighbors algorithm measures the distance to determine the “closeness” of instances.
```

File: curriculum/challenges/english/11-machine-learning-with-python/machine-learning-with-python-projects/book-recommendation-engine-using-knn.md [hunk 2]
```diff
@@ -8,16 +8,49 @@ dashedName: book-recommendation-engine-using-knn
...
+    ['Catch 22', 0.793983519077301], 
+    ['The Witching Hour (Lives of the Mayfair Witches)', 0.7448656558990479], 
+    ['Interview with the Vampire', 0.7345068454742432],
+    ['The Tale of the Body Thief (Vampire Chronicles (Paperback))', 0.5376338362693787],
+    ['The Vampire Lestat (Vampire Chronicles, Book II)', 0.5178412199020386]
+  ]
+]
+```
+
+Notice that the data returned from `get_recommends()` is a list. The first element in the list is the book title passed in to the function. The second element in the list is a list of five more lists. Each of the five lists contains a recommended book and the distance from the recommended book to the book passed in to the function.
```
Comment (Sembauke) [inline curriculum/challenges/english/11-machine-learning-with-python/machine-learning-with-python-projects/book-recommendation-engine-using-knn.md:48]:
```suggestion
Notice that the data returned from `get_recommends()` is a list. The first element in the list is the book title passed into the function. The second element in the list is a list of five more lists. Each of the five lists contains a recommended book and the distance from the recommended book to the book passed into the function.
```

File: curriculum/challenges/english/11-machine-learning-with-python/machine-learning-with-python-projects/cat-and-dog-image-classifier.md [hunk 3]
```diff
@@ -8,14 +8,98 @@ dashedName: cat-and-dog-image-classifier
...
+
+For the `*_data_gen` variables, use the `flow_from_directory` method. Pass in the batch size, directory, target size (`(IMG_HEIGHT, IMG_WIDTH)`), class mode, and anything else required. `test_data_gen` will be the trickiest one. For `test_data_gen`, make sure to pass in `shuffle=False` to the `flow_from_directory` method. This will make sure the final predictions stay is in the order that our test expects. For `test_data_gen` it will also be helpful to observe the directory structure.
+
+
+After you run the code, the output should look like this:
+
+```py
+Found 2000 images belonging to 2 classes.
+Found 1000 images belonging to 2 classes.
+Found 50 images belonging to 1 classes.
```
Comment (Sembauke) [inline curriculum/challenges/english/11-machine-learning-with-python/machine-learning-with-python-projects/cat-and-dog-image-classifier.md:56]:
```suggestion
Found 50 images belonging to 1 class.
```

File: curriculum/challenges/english/11-machine-learning-with-python/machine-learning-with-python-projects/cat-and-dog-image-classifier.md [hunk 4]
```diff
@@ -8,14 +8,98 @@ dashedName: cat-and-dog-image-classifier
...
+
+## Cell 10
+
+Now it is time to use your model to predict whether a brand new image is a cat or a dog.
+
+In this cell, get the probability that each test image (from `test_data_gen`) is a dog or a cat. `probabilities` should be a list of integers. 
+
+Call the `plotImages` function and pass in the test images and the probabilities corresponding to each test image.
+
+After your run the cell, you should see all 50 test images with a label showing the percentage sure that the image is a cat or a dog. The accuracy will correspond to the accuracy shown in the graph above (after running the previous cell). More training images could lead to a higher accuracy.
```
Comment (Sembauke) [inline curriculum/challenges/english/11-machine-learning-with-python/machine-learning-with-python-projects/cat-and-dog-image-classifier.md:97]:
```suggestion
After your run the cell, you should see all 50 test images with a label showing the percentage of "sure" that the image is a cat or a dog. The accuracy will correspond to the accuracy shown in the graph above (after running the previous cell). More training images could lead to a higher accuracy.
```

**Review by Sembauke — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/11-machine-learning-with-python/machine-learning-with-python-projects/rock-paper-scissors.md [hunk 5]
```diff
@@ -8,13 +8,57 @@ dashedName: rock-paper-scissors
...
+
+For development, you can use `main.py` to test your `calculate()` function. Click the "run" button and `main.py` will run.
+
+## Testing 
+
+We imported the tests from `test_module.py` to `main.py` for your convenience. The tests will run automatically whenever you hit the "run" button.
+
+## Submitting
+
+Copy your project's URL and submit it to freeCodeCamp.
```
Comment (moT01) [inline curriculum/challenges/english/11-machine-learning-with-python/machine-learning-with-python-projects/rock-paper-scissors.md:61]:
```suggestion
Copy your project's URL and submit it below.
```

**Review by moT01 — COMMENTED** [review]
This is looking good @nhcarrigan 👍 Here's a few things I noticed:

A few of the projects have comments in the cells or files that we won't be able to translate. I don't think this is a problem - they aren't all that important, but we could move them here if we want them translated. I would be fine leaving them how they are.

I noticed [a PR to remove the instructions from the cat and dog boilerplate.](https://github.com/freeCodeCamp/boilerplate-cat-and-dog-image-classifier/pull/4) Do we want to remove the instructions from all the boilerplates?

At the top of each notebook there's a short paragraph: `Note: You are currently reading this using Google Colaboratory...` Do we want to move that paragraph to these markdown files? Side note: that PR I linked removes that paragraph from the notebook.

^ This is the only one we may want to address on this PR. If we decide to leave that paragraph in the notebooks, I would be fine merging this. I left one suggestion in the code that's pretty minor.

**Comment by nhcarrigan** [comment]
> I noticed https://github.com/freeCodeCamp/boilerplate-cat-and-dog-image-classifier/pull/4 Do we want to remove the instructions from all the boilerplates?

The cat and dog one was the only notebook that had a bunch of instructions throughout the entire thing - the rest just have the initial paragraph.

**Comment by moT01** [comment]
Well, they mostly all have the instructions you moved here. The [book recommendation project](https://github.com/freeCodeCamp/boilerplate-book-recommendation-engine/blob/master/fcc_book_recommendation_knn.ipynb) for example, has all the same instructions that will be added here. You're saying we don't want to remove those?

**Review by Sembauke — APPROVED** [review]

**Comment by nhcarrigan** [comment]
> It's fine to leave them for now, just wondering if we want to or not.

I could remove the instructions, but we should probably leave the "watch this video about collab" link because it's a helpful resource and would make less sense on `/learn` IMHO.

**Review by moT01 — APPROVED** [review]
Alright, lets go with this then 🚀 I'll make an issue about removing instructions from the notebooks.
